src/main/java/seedu/address/logic/commands/AddContactCommand.java

```diff
@@ -77,9 +78,19 @@ private static InternshipApplication createInternshipWithContact(InternshipAppli
 
         CompanyName companyName = internshipToAddContact.getCompanyName();
         JobTitle jobTitle = internshipToAddContact.getJobTitle();
+        InternshipStatus status = internshipToAddContact.getStatus();
         InterviewDate interviewDate = internshipToAddContact.getInterviewDate();
 
-        return new InternshipApplication(companyName, jobTitle, contact, interviewDate);
+        if (status != null && interviewDate != null) {
+            return new InternshipApplication(companyName, jobTitle, contact, status, interviewDate);
+        } else if (status != null) {
+            return new InternshipApplication(companyName, jobTitle, contact, status);
+        } else if (interviewDate != null) {
+            return new InternshipApplication(companyName, jobTitle, contact, interviewDate);
+        } else {
+            return new InternshipApplication(companyName, jobTitle, contact);
+        }
+
```

Comment:
It seems like even if you are using different constructors, the corresponding missing value will still be set to null

Reply:
There's a function `requireAllNonNull` in the internship application call that doesn't allow null values from being passed in